### PR TITLE
WIP: Support for Windows Server Nano

### PR DESCRIPTION
**Pull request by TheFlyingCorpse**
This PR is a WIP, looking for feedback.

If Icinga2 on Nano is to be bundled as an "appx" which is the only supported package format in Nano, the following applies:
- No custom hooks or actions
- Only one install dir, not possible to copy or move files to other directories.
- The basic "users" check is not valid for Nano inherited from the default checks on windows, Terminal Services does not exist on Nano, there is no RDP, only remote PowerShell.

If not as an Appx, it just works, if it has something to generate its config, such as Puppet, Chef or Ansible.

The current state:
- It compiles and runs with MSVC12 and MSVC14 on Nano with the commits below.

Looking for feedback on how to best implement the following:
- Initialize Icinga2 skeleton.
-- Suggestion: What if the user must run a command after install, ie import a powershell module and run whatever powershell snippet they have to configure the agent? (see PowerShell commit for my suggestion on the skeleton)
- Configure icinga2
-- Icinga2 on windows cannot be a master, thus I dont think a lot of functionality is needed to get started with just configuring the basics which might be:
--- Local Zone (NodeName)
--- Local Endpoint (NodeName)
--- Upstream zone 
--- Upstream endpoints
--- Allow remote commands / config

Feedback most welcome.

Refs #4357

### CODE REVIEW

**Comment by TheFlyingCorpse** [comment]
I need to rebase this and remove the powershell commit.

**Comment by TheFlyingCorpse** [comment]
Removed powershell.

File: lib/base/scriptutils.cpp [hunk 1]
```diff
@@ -30,9 +30,12 @@
 #include <boost/regex.hpp>
 #include <algorithm>
 #include <set>
+
+#ifdef ICINGA2_WITH_MSI_FUNCTION
```
Comment (dnsmichi) [inline lib/base/scriptutils.cpp]: Is this really necessary? It looks like Nano would require different Windows binaries on its own. Is there a built-in define we could refer to, similar to _WIN32?

**Comment by dnsmichi** [comment]
Talked with @TheFlyingCorpse during OSMC - it seems that Microsoft changed the specs for Nano, and one cannot use it reliably for now. We'll have to wait for a new server release, or similar.

**Comment by dnsmichi** [comment]
I'll close this for now. Once there is a reliable way to use Nano servers, we'll continue investigating.

**Pull request closed without merging**